src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/JavaScriptExports.cs

```diff
@@ -192,6 +192,8 @@ public static void CompleteTask(JSMarshalerArgument* arguments_buffer)
             }
         }
 
+#if FEATURE_WASM_THREADS
```

Comment:
I think you need something like:
```xml
   <PropertyGroup>
     <FeatureWasmThreads Condition="'$(TargetsBrowser)' == 'true' and '$(WasmEnableThreads)' == 'true'">true</FeatureWasmThreads>
   </PropertyGroup>

   <PropertyGroup>
       <DefineConstants Condition="'$(FeatureWasmThreads)' == 'true'" >$(DefineConstants);FEATURE_WASM_THREADS</DefineConstants>
   </PropertyGroup>
```

in `src/libraries/System.Runtime.InteropServices.JavaScript/src/System.Runtime.InteropServices.JavaScript.csproj`
I don't think it inherits it from anywhere.

Reply:
Lesson painfully learned.  Don't use `$(TargetsBrowser)` in libraries projects.  They set `SkipInferTargetOSName` to `false`, and so `TargetsBrowser` is never set.  Instead, use `'$(TargetOS)' == 'browser'`, which does work.